lib/configparser.ex

```diff
@@ -266,8 +266,8 @@ defmodule ConfigParser do
   end
 
   # Returns true if the line appears to be a comment
-  @hash_comment_regex ~r{#.*}
-  @semicolon_comment_regex ~r{;.*}
+  @hash_comment_regex ~r{^#.*}
+  @semicolon_comment_regex ~r{^;.*}
```

Comment:
It might be clearer to incorporate the leading whitespace into these regexes, rather than use a separate call to `String.strip/1`. It keeps everything in one place and makes it a bit easier to reason about.